src/auth/auth0-client-provider.js

```diff
@@ -0,0 +1,20 @@
+import createAuth0Client from '@auth0/auth0-spa-js'
+import routes from '../routes'
+
+const auth0Options = {
+  domain: window.AUTH0_DOMAIN,
+  client_id: window.AUTH0_CLIENT_ID,
+  audience: window.AUTH0_AUDIENCE,
+  redirect_uri: `${window.location.origin}${routes.oauth}`
+}
+
+class Auth0ClientProvider {
```

Comment:
Why use a class here instead of just exporting the function?

Reply:
We're using the class to protect the way that the promise for the auth0 client is accessed (making it a singleton).